internal/pipe/mattermost/mattermost.go

```diff
@@ -0,0 +1,113 @@
+package mattermost
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/apex/log"
+	"github.com/caarlos0/env/v6"
+	"github.com/mattermost/mattermost-server/v5/model"
```

Comment:
seems like we're adding a lot of deps just to use the model package... maybe its better to copy that struct into our code instead?

Reply:
sounds great to me!

Reply:
will make the changes, thanks for your feedback and review!